src/mpid/ch4/shm/ipc/src/ipc_send.h

```diff
@@ -118,7 +118,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPCI_try_lmt_isend(const void *buf, MPI_Aint
     if (ipc_attr.gpu_attr.type == MPL_GPU_POINTER_DEV) {
         mpi_errno = MPIDI_GPU_get_ipc_attr(mem_addr, rank, comm, &ipc_attr);
         MPIR_ERR_CHECK(mpi_errno);
-    } else {
+    } else if (!MPL_gpu_query_pointer_is_dev(buf, &ipc_attr.gpu_attr)) {
```

Comment:
Isn't this already checked on line 116 and 118 above? `mem_addr` should point to the start of the data in `buf`.

Reply:
The rational behind this is that even host-registered memory (i.e. `zeMemAllocHost`) in ZE is device objects which XPMEM can't properly handle. So this check is to confirm `buf` is regular malloced memory. The other backends don't have this issue (I think), so we just check for `MPL_GPU_POINTER_DEV` in their implementations of `MPL_gpu_query_pointer_is_dev`.

Reply:
I see. And just to confirm memory from `zeMemAllocHost` doesn't support GPU IPC? I'm wondering if we could instead change the condition on line 118 to
```c
if (MPL_gpu_query_pointer_is_dev(buf, &ipc_attr.gpu_attr)) {
    mpi_errno = MPIDI_GPU_get_ipc_attr(mem_addr, rank, comm, &ipc_attr);
    MPIR_ERR_CHECK(mpi_errno);
}
```
If not, another comment here is probably needed to explain the peculiarity with ZE.

Reply:
I will add a comment about this peculiarity.

I am not fully updated on the current status, but the intention is that GPU IPC will eventually be supported for `zeMemAllocHost`. However, our current cache implementation does not handle this yet. Currently the caches are based on device ID, and host-registered memory does not have a device ID. So we would need to modify the cache to accept this type of memory before enabling it here.

Reply:
Added comments and reworded the commit messages slightly to reflect this discussion

Reply:
Looks good. Thanks.